### PR TITLE
Fixed the AzureTenantId in oauth.go

### PR DESCRIPTION
**Pull request by tubiskasarus**
AzureTenantId was hardcoded as a constant which was linked to ".staging.azuredatabricks.net". Therefore, if the host URL was for example from ".azuredatabricks.net", it failed to detect ClientId. I replaced the const value with a map to find the AzureTenantId based on the DSN host.

### CODE REVIEW

**Review by jadewang-db — APPROVED** [review]

**Comment by tubiskasaroos** [comment]
Apologies for submitting the pull request using my test account. I have now transferred the ownership of the forked repository to my main account. I wanted to clarify this to avoid any suspicion. 🙏

File: auth/oauth/oauth.go [hunk 1]
```diff
@@ -10,9 +10,18 @@ import (
 	"golang.org/x/oauth2"
 )
 
-const (
-	azureTenantId = "4a67d088-db5c-48f1-9ff2-0aace800ae68"
-)
+type AzureTenant struct {
+	DnsZone            string
+	AzureApplicationID string
+}
```
Comment (kravets-levko) [inline auth/oauth/oauth.go:16]: Is this type used anywhere?

Comment (tubiskasaroos) [reply auth/oauth/oauth.go:16]: No,  just removed it. Good catch.
At first I wanted to use it to replace the: 
`type CloudType int`
But I noticed that the InferCloudFromHost func in previous versions would be unstable. So I handled it separately.

**Comment by tubiskasaroos** [comment]
> @tubiskasaroos can you please rebase your branch? Looks it contains some irrelevant changes

I’m a bit apprehensive because this is my first time contributing on GitHub. I’ve pushed a change from a forked repository to the original one, and I attempted to rebase and address the previous issue related to signing off. However, I lack confidence in whether my actions were correct. Would it be possible to receive some assistance, or should I create a separate fork and submit another pull request?

**Comment by kravets-levko** [comment]
@tubiskasaroos sorry if my message feels a rude - there was no such intention at all. Of course I can help you.

Go to your fork, select `main` branch in dropdown. Under that dropdown there's a message "This branch is 2 commits behind databricks/databricks-sql-go:main", and next to it is a "Sync fork" button. Click it and select "Update branch".

Next, go to terminal/command line on your computer, and run these commands:

```bash
git checkut main
git pull
git checkout patch-1
git pull
git merge main
git push
```

There should be no conflicts, but in a case of any - resolve them.

I'll keep an eye on this PR. Please feel free to ping me if any questions.

<details><summary>Sync fork screenshot</summary>

image

</details>

**Comment by kravets-levko** [comment]
Much better now 🙂 Thank you!

**Comment by tubiskasaroos** [comment]
All good. Thank you so much for your help!

**Pull request merged**